src/components/radio-group/radio-group.tsx

```diff
@@ -124,36 +133,25 @@ export class RadioGroup extends React.Component<RadioGroupProps> {
     private createChildren(dataArray: React.ReactNode): React.ReactNode[] {
         return React.Children.map(dataArray, (child, index) => {
             if (child && typeof child === 'object') {
-                if (child.type === RadioButton) {
-                    return (
-                        <RadioButton
-                            key={index}
-                            value={child.props.value}
-                            data-automation-id={'RADIO_BUTTON_' + index}
-                            checked={this.checkedArray[index].checked}
-                            onChange={this.childrenOnClick(index)}
-                            disabled={this.props.disabled || child.props.disabled}
-                            readOnly={this.props.readOnly || child.props.readOnly}
-                            name={this.name}
-                            className="radioGroupChild"
-                            tabIndex={this.getChildTabIndex(index, this.isGroupChecked)}
-                            children={child.props.children}
-                        />
-                    );
-                } else {
-                    return (
-                        React.cloneElement(child,
-                            {
-                                key: index,
-                                checked: this.checkedArray[index].checked,
-                                onChange: action(this.childrenOnClick(index)),
-                                className: 'radioGroupChild',
-                                tabIndex: this.getChildTabIndex(index, this.isGroupChecked)
-                            },
-                            child.props.children
-                        )
-                    );
-                }
+
+                const childProps = {
+                    ...child.props,
+                    key: index,
+                    ...(child.type === RadioButton ?
```

Comment:
ummm... It will be more readable if you set aside this code to e.g. `extraProps` variable or something.